### PR TITLE
treewide: replace all pytest-cov patching outside of pythonPackages with pytest-cov-stub

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pbsds** [comment]
broken eval

File: pkgs/tools/backup/zfs-replicate/default.nix [hunk 1]
```diff
@@ -19,17 +19,14 @@ buildPythonApplication rec {
...
-  '';
-
   nativeBuildInputs = [
     poetry-core
   ];
 
   nativeCheckInputs = [
     pytestCheckHook
     hypothesis
+    pytest-cov-stub
```
Comment (mweinelt) [inline pkgs/tools/backup/zfs-replicate/default.nix:29]: not passed into the function

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 336777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li>ansible-later</li>
    <li>ansible-later.dist</li>
    <li>dep-scan</li>
    <li>dep-scan.dist</li>
    <li>makejinja</li>
    <li>makejinja.dist</li>
    <li>marge-bot</li>
    <li>marge-bot.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
    <li>telepresence</li>
    <li>telepresence.dist</li>
  </ul>
</details>
<details>
  <summary>44 packages built:</summary>
  <ul>
    <li>authentik</li>
    <li>autosuspend</li>
    <li>autosuspend.dist</li>
    <li>cmake-lint</li>
    <li>cmake-lint.dist</li>
    <li>codespell</li>
    <li>codespell.dist</li>
    <li>esphome</li>
    <li>esphome.dist</li>
    <li>gato</li>
    <li>gato.dist</li>
    <li>git-archive-all</li>
    <li>git-archive-all.dist</li>
    <li>git-pw</li>
    <li>git-pw.dist</li>
    <li>gpodder</li>
    <li>home-assistant-custom-components.govee-lan</li>
    <li>home-assistant-custom-components.mass</li>
    <li>ioc-scan</li>
    <li>ioc-scan.dist</li>
    <li>kb</li>
    <li>kb.dist</li>
    <li>kube-hunter</li>
    <li>kube-hunter.dist</li>
    <li>paperless-ngx</li>
    <li>pretalx</li>
    <li>pretalx.dist</li>
    <li>pretalx.static</li>
    <li>protonvpn-gui</li>
    <li>protonvpn-gui.dist</li>
    <li>refurb</li>
    <li>refurb.dist</li>
    <li>sshuttle</li>
    <li>sshuttle.dist</li>
    <li>sublime-music</li>
    <li>sublime-music.dist</li>
    <li>troubadix</li>
    <li>troubadix.dist</li>
    <li>trustymail</li>
    <li>trustymail.dist</li>
    <li>wapiti</li>
    <li>wapiti.dist</li>
    <li>zfs-replicate</li>
    <li>zfs-replicate.dist</li>
  </ul>
</details>


marge-bot:
```
ERROR: usage: __main__.py [options] [file_or_dir] [file_or_dir] [...]
__main__.py: error: unrecognized arguments: --flake8 --pylint
  inifile: /build/source/setup.cfg
  rootdir: /build/source
```

snapcraft:
```
Executing pytestCheckPhase
ERROR: Missing required plugins: pytest-cov>=4.0
```

the rest also fail on hydra

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 336777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>ansible-later</li>
    <li>ansible-later.dist</li>
    <li>dep-scan</li>
    <li>dep-scan.dist</li>
    <li>telepresence</li>
    <li>telepresence.dist</li>
  </ul>
</details>
<details>
  <summary>54 packages built:</summary>
  <ul>
    <li>authentik</li>
    <li>autosuspend</li>
    <li>autosuspend.dist</li>
    <li>cmake-lint</li>
    <li>cmake-lint.dist</li>
    <li>codespell</li>
    <li>codespell.dist</li>
    <li>esphome</li>
    <li>esphome.dist</li>
    <li>gato</li>
    <li>gato.dist</li>
    <li>git-archive-all</li>
    <li>git-archive-all.dist</li>
    <li>git-pw</li>
    <li>git-pw.dist</li>
    <li>gpodder</li>
    <li>home-assistant-custom-components.govee-lan</li>
    <li>home-assistant-custom-components.mass</li>
    <li>ioc-scan</li>
    <li>ioc-scan.dist</li>
    <li>kb</li>
    <li>kb.dist</li>
    <li>kube-hunter</li>
    <li>kube-hunter.dist</li>
    <li>makejinja</li>
    <li>makejinja.dist</li>
    <li>marge-bot</li>
    <li>marge-bot.dist</li>
    <li>paperless-ngx</li>
    <li>pretalx</li>
    <li>pretalx.dist</li>
    <li>pretalx.static</li>
    <li>protonvpn-gui</li>
    <li>protonvpn-gui.dist</li>
    <li>python311Packages.typed-settings</li>
    <li>python311Packages.typed-settings.dist</li>
    <li>python312Packages.typed-settings</li>
    <li>python312Packages.typed-settings.dist</li>
    <li>refurb</li>
    <li>refurb.dist</li>
    <li>snapcraft</li>
    <li>snapcraft.dist</li>
    <li>sshuttle</li>
    <li>sshuttle.dist</li>
    <li>sublime-music</li>
    <li>sublime-music.dist</li>
    <li>troubadix</li>
    <li>troubadix.dist</li>
    <li>trustymail</li>
    <li>trustymail.dist</li>
    <li>wapiti</li>
    <li>wapiti.dist</li>
    <li>zfs-replicate</li>
    <li>zfs-replicate.dist</li>
  </ul>
</details>

**Comment by pbsds** [comment]
i relaxed `appthreat-vulnerability-db` in dep-scan and confirmed that `--cov` is accepted by pytest

**Pull request merged**